avm/src/vm/event.rs

```diff
@@ -248,36 +259,44 @@ impl HandlerFragment {
     &mut self,
     mut hand_mem: Arc<HandlerMemory>,
     instrs: &Vec<Instruction>,
-  ) -> Arc<HandlerMemory> {
+  ) -> VMResult<Arc<HandlerMemory>> {
     if instrs.len() == 1 {
       let op = &instrs[0];
       if let OpcodeFn::Io(func) = op.opcode.fun {
         //eprintln!("{} {:?}", op.opcode._name, op.args);
         return func(op.args.clone(), hand_mem).await;
       } else {
-        eprintln!("expected an IO instruction");
-        std::process::exit(1);
+        return Err(VMError::InvalidState(InvalidState::UnexpectedInstruction(
+          InstrType::IO,
+        )));
       }
     } else {
       let futures: Vec<_> = instrs
         .iter()
         .map(|i| {
-          if let OpcodeFn::Io(func) = i.opcode.fun {
-            //eprintln!("{} {:?}", i.opcode._name, i.args);
-            func(i.args.clone(), HandlerMemory::fork(hand_mem.clone()))
-              .then(HandlerMemory::drop_parent_async)
-          } else {
-            eprintln!("expected another IO instruction");
-            std::process::exit(1);
+          let hand_mem = hand_mem.clone();
+          async move {
+            if let OpcodeFn::Io(func) = i.opcode.fun {
+              //eprintln!("{} {:?}", i.opcode._name, i.args);
```

Comment:
These comments are for future debugging?

Reply:
yes - I didn't remove it since it's been there for a while (and could prove useful in the future)